lib/ValueAnonymizers/HashService/StringHashSha256.php

```diff
@@ -7,18 +7,13 @@
 final class StringHashSha256 implements StringHashInterface
 {
 
-    private static string $salt;
+    public static string $salt;
```

Comment:
I'm thinking we can keep this private as it does not need to be accessed from outside the class.

With a setter that can only set it once, to respect the design that we have once one value for a run.

Reply:
And maybe we can keep using the salt initialization from constructor.
I'm not so sure how this could be used/extended by others but if they have another way to configure the app and not set the salt, we can at least make sure we initialize it...

Reply:
Regarding the salt initialization in the constructor, it was removed because we already have a default value for salt (see lib/Application/CommandLineParameters.php:49) set to its previous value (md5 of microtime()). Hence, the salt value will always be initialized even if not explicitly included in the command line. Or maybe I didn't understand correctly what you meant?

Reply:
I made salt private again and added a setter in ba3bb9f.

Reply:
Thanks, I think it should be enough.
I mean, it's not ideal, it's already a state that we manage and we shouldn't.
Application wise, it has no big impact as it is as the logic is simple but in general, you know, maintaining unnecessary state is bad.

The point was that if someone would use the application as a library, they might not use the cli part and configure things manually.
Until now we had only a global state that can only be set once.
Now we have a global state that can be set multiple times and once we might have other places in code (not necessary our code) that would set it, it will not be a good thing and bugs might sneak in.

The better approach, I think it would be to:
- pass the salt to `InterfaceProviderBuilder::buildProvider()` in `Setup::getAnonymizationProvider()`
- pass the salt to `ValueAnonymizerFactoryInterface::getValueAnonymizerClass()` in `YamlProviderBuilder::buildProvider()`
- pass the salt to `StringHashSha256::__construct()` in `ValueAnonymizerFactory::getValueAnonymizerClass()` and store it.
- pass the salt to `Eav::__construct()` in `ValueAnonymizerFactory::getValueAnonymizerClass()` and store it.
- pass the salt to `StringHashSha256::__construct()` in `Eav::anonymize()` and store it.

But I think it might not be worth it that much...
We have other parts that we can improve so let's leave it as it is for now.